datamap/mv/plan/src/main/scala/org/apache/carbondata/mv/plans/util/MVSQLOptimizer.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.carbondata.mv.plans.util
+
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.rules.RuleExecutor
+
+// used to optimize the mv sql
+object MVSQLOptimizer extends RuleExecutor[LogicalPlan] {
```

Comment:
Why there is a need to add new class to add this optimizer, we already have a class classed BirdcageOptimizer right, where all the rule Batches are added, can't we use the same for this rule? so that, before going for modular plan, during optimization of logical plan, this rule will also be applied.

Reply:
Because this  ```MVAddGroupByColsToProjection``` is only used by MV SQL,  we shouldn't put the ```MVAddGroupByColsToProjection``` in BirdcageOptimizer to be used in other scene

Reply:
BirdcageOptimizer  is specifically for MVsession right, it will be called to optimize during MV queries only right

Reply:
@akashrn5 no， some class doesn't use BirdcageOptimizer,  such as MVDataMapProvider and MVHelper which this PR modify